### PR TITLE
Updated release notes for 3.2

### PR DESCRIPTION
**Pull request by fonnesbeck**
Please check for accuracy and completeness. 

I will test the notebooks and examples and check the installation in a clean env.

### CODE REVIEW

File: RELEASE-NOTES.md [hunk 1]
```diff
@@ -1,5 +1,62 @@
...
+### New features
+
+This version includes two major contributions from our Google Summer of Code 2017 students:
+
+- Maxim Kucherov extended and refactored the variational inference module. This primarily adds two important classes, representing operator variational inference (`OPVI`) objects and `Approximation` objects. These make it easier to extend existing `variational` classes, and to derive inference from `variational` optimizations, respectively. The `variational` module now also includes normalizing flows (`NFVI`).
+- William Engels added an extensive new Gaussian processes (`gp`) module. Standard GPs can be specified using either `Latent` or `Marginal` classes, depending on the nature of the underlying function. A Student-T process `TP` has been added. In order to accomodate larger datasets, approximate marginal Gaussian processes (`MarginalSparse`) have been added. 
+
+Documentation has been improved as the result of the project's monthly "docathons".
+
+An experimental stochastic gradient Fisher scoring (`SGFS`) class has been added to the `variational` module.
```
Comment (junpenglao) [inline RELEASE-NOTES.md]: it's not a variational method, it is a MCMC sampler.

**Comment by aseyboldt** [comment]
Should we also add a section about the theano end of life issue?

**Comment by fonnesbeck** [comment]
@aseyboldt I don't think so. Separate issue from the 3.2 release details.

**Comment by fonnesbeck** [comment]
Local test results:

    1634 passed, 14 skipped, 66 warnings

**Comment by bwengals** [comment]
Could i be bill instead of william?

File: RELEASE-NOTES.md [hunk 2]
```diff
@@ -1,5 +1,62 @@
...
+
+QuadPotential adaptation has been implemented.
+
+Script added to build and deploy documentation.
+
+MAP estimates now available for transformed and non-transformed variables.
+
+The `Constant` variable class has been deprecated, and will be removed in 3.3.
+
+DIC an BPIC calculations have been sped up.
```
Comment (ColCarroll) [inline RELEASE-NOTES.md]: typo:
> DIC **and** BPIC

File: RELEASE-NOTES.md [hunk 3]
```diff
@@ -1,5 +1,62 @@
...
+This version includes two major contributions from our Google Summer of Code 2017 students:
+
+- Maxim Kucherov extended and refactored the variational inference module. This primarily adds two important classes, representing operator variational inference (`OPVI`) objects and `Approximation` objects. These make it easier to extend existing `variational` classes, and to derive inference from `variational` optimizations, respectively. The `variational` module now also includes normalizing flows (`NFVI`).
+- Bill Engels added an extensive new Gaussian processes (`gp`) module. Standard GPs can be specified using either `Latent` or `Marginal` classes, depending on the nature of the underlying function. A Student-T process `TP` has been added. In order to accomodate larger datasets, approximate marginal Gaussian processes (`MarginalSparse`) have been added. 
+
+Documentation has been improved as the result of the project's monthly "docathons".
+
+An experimental stochastic gradient Fisher scoring (`SGFS`) sampling step method has been added.
+
+The API for find_MAP was enhanced.
```
Comment (ColCarroll) [inline RELEASE-NOTES.md]: backticks around `find_MAP`

File: RELEASE-NOTES.md [hunk 4]
```diff
@@ -1,5 +1,62 @@
...
+
+Script added to build and deploy documentation.
+
+MAP estimates now available for transformed and non-transformed variables.
+
+The `Constant` variable class has been deprecated, and will be removed in 3.3.
+
+DIC an BPIC calculations have been sped up.
+
+Arrays are now accepted as arguments for the `Bound` class.SGFS
```
Comment (ColCarroll) [inline RELEASE-NOTES.md]: is `SGFS` at the end here a typo?

**Comment by fonnesbeck** [comment]
Installation via pip into a clean env works here. Just spot checking a few notebooks, and we should be good to merge and tag the RC.

**Comment by fonnesbeck** [comment]
If someone can give a thumbs up, I will merge when Travis passes.

**Comment by ColCarroll** [comment]
Looks good to me -- I'm running all the notebooks locally on master, too.

**Comment by fonnesbeck** [comment]
Ok. That takes a long time, by the way.

**Comment by ColCarroll** [comment]
Background task + [my own CLI](https://pypi.org/project/carpo/).  I can respond later if anything fails!

screen shot 2017-10-02 at 2 23 21 pm

File: RELEASE-NOTES.md [hunk 5]
```diff
@@ -1,5 +1,62 @@
...
+## PyMC3 3.2 (October 2, 2017)
+
+### New features
+
+This version includes two major contributions from our Google Summer of Code 2017 students:
+
+- Maxim Kucherov extended and refactored the variational inference module. This primarily adds two important classes, representing operator variational inference (`OPVI`) objects and `Approximation` objects. These make it easier to extend existing `variational` classes, and to derive inference from `variational` optimizations, respectively. The `variational` module now also includes normalizing flows (`NFVI`).
+- Bill Engels added an extensive new Gaussian processes (`gp`) module. Standard GPs can be specified using either `Latent` or `Marginal` classes, depending on the nature of the underlying function. A Student-T process `TP` has been added. In order to accomodate larger datasets, approximate marginal Gaussian processes (`MarginalSparse`) have been added. 
+
+Documentation has been improved as the result of the project's monthly "docathons".
```
Comment (twiecki) [inline RELEASE-NOTES.md]: We used `*` bullet-points beneath.

Comment (twiecki) [inline RELEASE-NOTES.md]: just "docathon" I think

Comment (fonnesbeck) [reply RELEASE-NOTES.md]: Didnt we have two?

**Comment by fonnesbeck** [comment]
@ColCarroll let me know when your tests are done.

**Comment by ColCarroll** [comment]
As you pointed out, it could be a while -- just scanning `GLM-Logistic` there are two samplers that take ~30mins each.

**Comment by fonnesbeck** [comment]
⏳

**Comment by ColCarroll** [comment]
`GLM-logistic` took 3,144s in the end!  Still working through the `GLM-*`'s, but won't finish running them all today.  Only bug I've found so far is a tiny one in `GLM-Latent`, which is missing `import sys`.

**Comment by ColCarroll** [comment]
In `GP-MaunaLoa` 

```
predict_region = BoxAnnotation(left=pd.to_datetime("2003-12-15"), 
                                                      fill_alpha=0.1, fill_color="firebrick")
```

throws an error because it thinks `left` should be a string, not a `datetime` (I can submit a PR later tonight fixing these two).

**Comment by fonnesbeck** [comment]
@ColCarroll any update on this?

**Comment by ColCarroll** [comment]
Might need @bwengals to check on/re-render the GP-MaunaLoa.  I'm getting an error on passing `pd.datetime` to a `BoxAnnotation` object, but haven't had time to mess with finding the right version of Bokeh.  

Also, GP-slice-sampling looks like it has an actual bug ([here's the trace](https://gist.github.com/ColCarroll/eb62dc2cefd593026d1d22a1b4e5f4aa)). 

Got hung up on PyMC3_tips_and_heuristic (let it run for ~2hrs), but going to manually skip and run the rest, hopefully!

**Comment by junpenglao** [comment]
> Got hung up on PyMC3_tips_and_heuristic (let it run for ~2hrs)

Wow, that's long. I think we should set the init to advi there .

**Comment by junpenglao** [comment]
@ColCarroll yep just run it locally, the first two models are extremely slow during tuning (the first model is kind of a proof of concept tho, in the doc I advice *not* to do it that way). I can send a PR to add a kwarg.

**Comment by fonnesbeck** [comment]
Can we move forward with the RC without worrying about the runtime?

@ColCarroll were there other notebook bugs?

**Comment by ColCarroll** [comment]
None beyond the two flagged above -- I might have to re-jigger that CLI to deal better with long running notebooks.

**Pull request merged**